config/runtimes/tgis-template.yaml

```diff
@@ -0,0 +1,59 @@
+apiVersion: template.openshift.io/v1
+kind: Template
+metadata:
+  labels:
+    opendatahub.io/dashboard: 'true'
+    opendatahub.io/ootb: 'true'
+  annotations:
+    description: Text Generation Inference Server (TGIS) is a high performance inference engine that deploys and serves Large Language Models.
+    openshift.io/display-name: TGIS Standalone ServingRuntime for KServe
+    openshift.io/provider-display-name: Red Hat, Inc.
+    tags: rhods,rhoai,kserve,servingruntime
+    template.openshift.io/documentation-url: https://github.com/opendatahub-io/text-generation-inference
+    template.openshift.io/long-description: This template defines resources needed to deploy TGIS standalone servingruntime with KServe in Red Hat OpenShift AI
+    opendatahub.io/modelServingSupport: '["single"]'
+    opendatahub.io/apiProtocol: 'gRPC'
+  name: tgis-grpc-serving-template
+objects:
+  - apiVersion: serving.kserve.io/v1alpha1
+    kind: ServingRuntime
+    metadata:
+      name: tgis-grpc-runtime
+      annotations:
+        openshift.io/display-name: TGIS Standalone ServingRuntime for KServe
+        opendatahub.io/recommended-accelerators: '["nvidia.com/gpu"]'
+      labels:
+        opendatahub.io/dashboard: 'true'
+    spec:
+      multiModel: false
+      supportedModelFormats:
+        - autoSelect: true
+          name: pytorch
+      containers:
+        - name: kserve-container
+          image: $(tgis-image)
+          command: ['text-generation-launcher']
+          args:
+            - '--model-name=/mnt/models/'
+            - '--port=3000'
+            - '--grpc-port=8033'
+          env:
+            - name: TRANSFORMERS_CACHE
+              value: /tmp/transformers_cache
+          readinessProbe:
+            exec:
+              command:
+                - curl
+                - localhost:3000/health
+            initialDelaySeconds: 5
```

Comment:
Should we make `initialDelaySeconds` a parameter?

This was added in caikit-tgis-serving templates: https://github.com/opendatahub-io/caikit-tgis-serving/pull/252

Here's an example of what this could look like with kustomize: https://github.com/heyselbi/odh-model-controller/compare/move-templates...dtrifiro:odh-model-controller:move-templates (from by branch [here](https://github.com/dtrifiro/odh-model-controller/tree/move-templates).

We could do this in another PR if you don't want to change this though.

Reply:
Would it be possible to move this line https://github.com/heyselbi/odh-model-controller/compare/move-templates...dtrifiro:odh-model-controller:move-templates#diff-c0e66b0ce356fbd36d9261b053a4c40c647efa33cf0870a2a0d14bc00940a396R6 to a patch instead?

Reply:
@Xaenalt @Jooho if that's okay for you too, I can add it to this PR.

Reply:
@heyselbi I'm, not sure what you mean here.

Reply:
@dtrifiro What's the rationale behind parameterizing it? Are there particular motivations for centralizing control over this aspect? Should this parameter vary each time the odh-model-controller is installed or updated?
If so, I recommend using `patch` from customization.yaml file.

Reply:
Starting a `ServingRuntime` takes a different amount of time depending on the model size: it takes time to copy the model and to load it into memory. QE had been seeing deployment errors with large model size.

The initial idea was to modify the templates so that `initialDelaySeconds` could be modified by the user from the dashboard.

I have no strong opinions on how this is implemented (`kustomization.yaml` or patch), as long as it's possible for the final user to instantiate deployments from these templates with customized values of `initialDelaySeconds`)

Reply:
If you want to keep it parameterized i'd advice putting a placeholder/default value here and working with UI to expose the parameter and have the dashboard apply it when creating the resource from the template

Reply:
@heyselbi I think this should do https://github.com/opendatahub-io/odh-model-controller/commit/25f205e087e95ece9b525530bfa4421f8586dc9f

Reply:
Done added it.